api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -307,11 +312,16 @@ public static Builder builderFor(Schema schema) {
     private final Set<String> partitionNames = Sets.newHashSet();
     private Map<Integer, PartitionField> timeFields = Maps.newHashMap();
     private int specId = 0;
+    private int lastAssignedFieldId = PARTITION_DATA_ID_START - 1;
```

Comment:
We tend to prefer using `AtomicInteger` instead of an `int` field. That way it is more clear because the calls are `incrementAndGet` or `getAndIncrement`. That would also avoid the need to start at `PARTITION_DATA_ID_START - 1`. You could use the constant to initialize and assign using `getAndIncrement`.

Reply:
I think we still have to use `PARTITION_DATA_ID_START - 1` to handle the case that there is no assigned field Id. Or we can update this field as `nextFieldId` and then set it from `PARTITION_DATA_ID_START`.